partiql-value/src/lib.rs

```diff
@@ -247,11 +263,23 @@ impl BinaryOr for Value {
     }
 }
 
+impl ops::Not for &Value {
+    type Output = Value;
+
+    fn not(self) -> Self::Output {
+        match self {
+            Value::Boolean(b) => Value::from(!b),
+            Value::Null | Value::Missing => Value::Null,
+            _ => Value::Missing, // data type mismatch => Missing
+        }
+    }
+}
+
 impl ops::Not for Value {
```

Comment:
Should we delete the `ops::Not` trait that works on `Value` (not reference)? Or should both be kept?